ADA/databricks_rstudio_sql_warehouse.qmd

```diff
@@ -1,105 +1,255 @@
 
-        ![](../images/databricks-test-connection.png)
+```         
+DATABRICKS_HOST=adb-5037484389568426.6.azuredatabricks.net
+DATABRICKS_SQL_WAREHOUSE_ID=<sql-warehouse-id>
+DATABRICKS_TOKEN=<personal-access-token> 
+DATABRICKS_CLUSTER_ID=<personal-cluster-id>
+```
 
-        d.  Click the 'OK' button to exit the 'Test Results' window, then the 'OK' button in the 'Simba Spark ODBC Driver DSN Setup' window.
+Once you have entered the details, save and close your .Renviron file.
 
-5.  Connect through RStudio. Watch the below video and view the [ADA_RStudio_connect GitHub repo](https://github.com/dfe-analytical-services/ADA_RStudio_connect) for methods on connecting to Databricks and querying data from RStudio. 
+::: callout-note
+Everyone in your team that wishes to connect to the SQL Warehouse in Databricks and run your code must set up their .Renviron file individually, otherwise their connection will fail.
+:::
 
+The sections below describe where to find the information needed for each of the four environment variables.
 
 ------------------------------------------------------------------------
 
-# Pulling data into R studio from Databricks
+#### Databricks host
+
+------------------------------------------------------------------------
+
+The Databricks host is the instance of Databricks that you want to connect to. It's the URL that you see in your browser bar when you're on the Databricks site and should end in "azuredatabricks.net".
+
+------------------------------------------------------------------------
+
+#### Databricks SQL Warehouse ID
+
+------------------------------------------------------------------------
+
+The Databricks SQL warehouse ID is the warehouse containing data that you would like to use. To get the warehouse ID, follow these steps:
+
+-   click 'SQL Warehouses' under the 'SQL' section of the left hand menu on Databricks
+-   click on the warehouse name that you'd like to get the ID for
+-   the warehouse ID is in brackets next to the warehouse name on the 'Overview' tab
 
 ------------------------------------------------------------------------
 
-Once you have set up an ODBC connection as detailed above, you can then use that connection to pull data directly from Databricks into R Studio. Charlotte recorded a video demonstrating two possible methods of how to do this. The recording is embedded below: 
+#### Databricks Token
+
+------------------------------------------------------------------------
+
+The Databricks token is the personal access token we generated in the [Get a personal access token from Databricks for authentication](/ADA/databricks_rstudio_sql_warehouse.html#get-a-personal-access-token-from-databricks-for-authentication) section.
+
+------------------------------------------------------------------------
+
+#### Cluster ID
+
+------------------------------------------------------------------------
+
```

Comment:
We could also add this example for when someone wants to connect with compute cluster rather than SQL Warehouse:

```r
library(tidyverse)
library(sparklyr)
library(dbplyr)

sc <- spark_connect(
  cluster_id = Sys.getenv("DATABRICKS_CLUSTER_ID"),
  method = "databricks_connect"
)

tbl(
  sc,
  I("catalog_10_gold.information_schema.catalogs")
) |>
  head()
```

The only difference when working locally as far as I can tell is that compute clusters can read files from Volumes,  whereas ODBC can only interact with tables. So with `sparklyr` and personal compute you can do:

```r
df <- spark_read_csv(
  sc,        
  "/Volumes/catalog_40_copper/folder/file.csv"
)
```

Reply:
I've added this - thanks!

Reply:
I'd avoid mentioning sparklyr in this page. It wants it's own guidance page really, and this method works just as well for personal clusters as it does for SQL warehouses.